File: valid-palindrome/valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        if (s.length() == 1) {
            return true;
        }
             
        for (int n = 0; n < s.length(); n++){
            if (isalnum(s[n]) == 0) {
                s.erase(s.begin()+n);
                n--;
            }
        }
        
        int i = 0;
        int j = (s.length() - 1);
        
        while (i < j) {
            s[i] = tolower(s[i]);
            s[j] = tolower(s[j]);
            if (s[i] != s[j]){
                return false;
            }
            
            i++;
            j--;            
        }
        
        return true;
    }
};

File: valid-palindrome/valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        
        #tolower s
        s = s.lower()
        
        #cycle through each letter in s and remove all letters not alphanumeric
        for l in s:
            if not l.isalnum():
                #replaces all instances of character l in s with "", not just those in the current position 
                s = s.replace(l, "")
        
        end = len(s) - 1
        for index, l in enumerate(s):
            if index >= (end - index):
                break #we have checked all values, met in center 
            if s[index] != s[end - index]:
                return False
        
        return True